lib/api/api_auth.go

```diff
@@ -203,6 +205,26 @@ func createSession(cookieName string, username string, guiCfg config.GUIConfigur
 	emitLoginAttempt(true, username, r.RemoteAddr, evLogger)
 }
 
+func handleLogout(cookieName string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		cookie, err := r.Cookie(cookieName)
+		if err == nil && cookie != nil {
+			sessionsMut.Lock()
+			delete(sessions, cookie.Value)
+			sessionsMut.Unlock()
+		}
```

Comment:
Does it make sense to continue as if it was a success if the condition isn't met (resp. the cookie doesn't exist in sessions)? Maybe related to my question above about this method being part of `noauth`, as if it required auth to call logout the user would get a 403 if the cookie wasn't present (which is what I'd have expected, not saying that's right, just gut feeling).

Reply:
I figure if the condition isn't met, then the session has already expired on the server side, so there's no point in keeping the cookie, so you might as well clean it up if the cookie is still there. Either way, from the user's point of view, the result is a successful logout operation, even if there wasn't really anything to do.

As I write this I realize this directly contradicts my conclusion in [the other thread](https://github.com/emlun/syncthing/pull/2/files#r1271457521)... :joy:

Reply:
> Either way, from the user's point of view, the result is a successful logout operation, even if there wasn't really anything to do.

I mean, say you attempt to log out but your session has already expired, so the API hits you with a 403 response. So you'd have to log in again before you can successfully log out... that wouldn't make much sense as a user experience, would it?

Reply:
Yeah, that makes sense.